File: Data Structures & Algorithms/longest-common-prefix/submission-0.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.len() < 1 {
            return String::new();
        }
        for (i, &ch) in (&strs[0]).as_bytes().iter().enumerate() {
            for s in &strs[1..] {
                if i >= s.len() || s.as_bytes()[i] != ch {
                    return String::from(&strs[0][0..i]);
                }
            }
        }
        return String::from(&strs[0]);
    }
}

File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-0.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int m = 0;
        int n = 0;
        while (m < s.size() && n < t.size()) {
            if (s[m] == t[n]) {
                n ++;
            }
            m ++;
        }
        if (n == t.size()) {
            return 0;
        } else return (t.size() - n);
    }
};